File: CMS/src/app/shared/CompanyInfoVungTrongNongHo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { CompanyInfoVungTrongNongHo } from 'src/app/shared/CompanyInfoVungTrongNongHo.model';
import { BaseService } from './Base.service';
@Injectable({
    providedIn: 'root'
})
export class CompanyInfoVungTrongNongHoService extends BaseService{

    List: CompanyInfoVungTrongNongHo[] | undefined;
    ListFilter: CompanyInfoVungTrongNongHo[] | undefined;
    FormData!: CompanyInfoVungTrongNongHo;

    DisplayColumns001: string[] = ['ThanhVienID','CCCD','DienThoai','Email','NamSinh','DiaChi','KinhDo','ViDo','Giong','NamTrong','ChungNhanVietGap', 'Save'];

    constructor(public httpClient: HttpClient) {
        super(httpClient);
        this.Controller = "CompanyInfoVungTrongNongHo";
    }
}

File: Service/Implement/CompanyInfoVungTrongNongHoService.cs
using Service.Interface;

namespace Service.Implement
{
    public class CompanyInfoVungTrongNongHoService : BaseService<CompanyInfoVungTrongNongHo, ICompanyInfoVungTrongNongHoRepository>
    , ICompanyInfoVungTrongNongHoService
    {
        private readonly ICompanyInfoVungTrongNongHoRepository _CompanyInfoVungTrongNongHoRepository;

        private readonly IThanhVienService _ThanhVienService;

        public CompanyInfoVungTrongNongHoService(ICompanyInfoVungTrongNongHoRepository CompanyInfoVungTrongNongHoRepository
            , IThanhVienService thanhVienService
        ) : base(CompanyInfoVungTrongNongHoRepository)
        {
            _CompanyInfoVungTrongNongHoRepository = CompanyInfoVungTrongNongHoRepository;

            _ThanhVienService = thanhVienService;   
        }
        public override void Initialization(CompanyInfoVungTrongNongHo model)
        {
            if (model.ThanhVienID > 0)
            {
                ThanhVien thanhVien = _ThanhVienService.GetByID(model.ThanhVienID.Value);
                model.Name = thanhVien.Name;
                model.CCCD = thanhVien.CCCD;
                model.DienThoai = thanhVien.DienThoai;
                model.Email = thanhVien.Email;
            }
        }
    }
}

